x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/whitelist/InternalSqlScriptUtils.java

```diff
@@ -27,110 +38,323 @@
+    }
+
+    public static Double asin(Number value) {
+        return MathOperation.ASIN.apply(value);
+    }
+
+    public static Double atan(Number value) {
+        return MathOperation.ATAN.apply(value);
+    }
+
+    public static Double cbrt(Number value) {
+        return MathOperation.CBRT.apply(value);
+    }
+
+    public static Double ceil(Number value) {
+        return MathOperation.CEIL.apply(value);
+    }
+
+    public static Double cos(Number value) {
+        return MathOperation.COS.apply(value);
+    }
+
+    public static Double cosh(Number value) {
+        return MathOperation.COSH.apply(value);
+    }
+
+    public static Double cot(Number value) {
+        return MathOperation.COT.apply(value);
+    }
+
+    public static Double degrees(Number value) {
+        return MathOperation.DEGREES.apply(value);
+    }
+
+    public static Double e(Number value) {
+        return MathOperation.E.apply(value);
+    }
+
+    public static Double exp(Number value) {
+        return MathOperation.EXP.apply(value);
+    }
+
+    public static Double expm1(Number value) {
+        return MathOperation.EXPM1.apply(value);
+    }
+
+    public static Double floor(Number value) {
+        return MathOperation.FLOOR.apply(value);
+    }
+
+    public static Double log(Number value) {
+        return MathOperation.LOG.apply(value);
+    }
+
+    public static Double log10(Number value) {
+        return MathOperation.LOG10.apply(value);
+    }
+
+    public static Double pi(Number value) {
+        return MathOperation.PI.apply(value);
+    }
+
+    public static Double radians(Number value) {
+        return MathOperation.RADIANS.apply(value);
+    }
+
+    public static Double random(Number value) {
+        return MathOperation.RANDOM.apply(value);
+    }
+
+    public static Double sign(Number value) {
+        return MathOperation.SIGN.apply(value);
+    }
+
+    public static Double sin(Number value) {
+        return MathOperation.SIN.apply(value);
+    }
+
+    public static Double sinh(Number value) {
+        return MathOperation.SINH.apply(value);
+    }
+
+    public static Double sqrt(Number value) {
+        return MathOperation.SQRT.apply(value);
+    }
+
+    public static Double tan(Number value) {
+        return MathOperation.TAN.apply(value);
+    }
+
+    //
+    // Date/Time functions
+    //
+    public static Integer dateTimeChrono(Object dateTime, String tzId, String chronoName) {
+        if (dateTime == null || tzId == null || chronoName == null) {
+            return null;
+        }
+        // use dedicated method
+        if (dateTime instanceof JodaCompatibleZonedDateTime) {
+            return DateTimeFunction.dateTimeChrono(((JodaCompatibleZonedDateTime) dateTime).getZonedDateTime(), tzId, chronoName);
+        }
+        throw new SqlIllegalArgumentException("Invalid date encountered [{}]", dateTime);
+    }
+    
+    public static String dayName(Long millis, String tzId) {
```

Comment:
Testing `select day_name(birth_date) from test_emp group by day_name(birth_date);` throws a NPE. Using the `_translate` API the error is summed up as
```
          "reason": {
                    "type": "script_exception",
                    "reason": "runtime error",
                    "script_stack": [
                        "InternalSqlScriptUtils.dayName(InternalSqlScriptUtils.docValue(doc,params.v0).millis, params.v1)",
                        "                                                                             ^---- HERE"
                    ],
                    "script": "InternalSqlScriptUtils.dayName(InternalSqlScriptUtils.docValue(doc,params.v0).millis, params.v1)",
                    "lang": "painless",
                    "caused_by": {
                        "type": "null_pointer_exception",
                        "reason": null
                    }
                }
```

Reply:
Looks like the current test suite didn't catch this. I've added some tests for day_name, month_name and quarter as they all suffer from the same problem.